### DISCUSSION TITLE
How can I include new attribute before render

### DISCUSSION THREAD

**Discussion by erbsutil** [remarkjs/remark#1018] [General] [answered]
I want to include a new tag, for example: 'rel', but only before rendering on my screen and using react-markdown.

Is that possible?

**Comment by ChristianMurphy** [comment, marked as answer]
Both new tags and new attributes can be added in react-markdown.
A few ways that could be handled.
* add a component handler for the HTML tag you want to add/modify https://github.com/remarkjs/react-markdown#use-custom-components-syntax-highlight
* Or add a rehype (html AST) plugin to tranform the tag or attribute https://unifiedjs.com/learn/

**Reply by erbsutil** [reply]
Hi, @ChristianMurphy!

Sorry i wrote wrong, I want to insert a new attribute inside 'a' (links)... Sorry, but I already reading your links, thanks!!!

**Reply by ChristianMurphy** [reply]
It is possible and the links above are still relevant

**Reply by erbsutil** [reply]
@ChristianMurphy "#use-custom-components-syntax-highlight", is directly to insert attributs on links in a MD code? I couldn’t understood how it works yet :\

**Reply by ChristianMurphy** [reply]
It shows how you can use a react element to render an HTML tag.
Which in your case, would be adding a component for the `a` tag and adding the attribute(s) you want.

**Reply by erbsutil** [reply]
@ChristianMurphy It worked better for me: https://github.com/remarkjs/react-markdown#appendix-b-components

For example: 

```
<ReactMarkdown
  children="[example](http://www.example.com)"
  components={{
    a: ({ node, ...props }) => (
      <a {...props} rel="noreferrer" />
    ),
  }}
/>
```

Thanks a lot!
